lib/ret/media_search.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Ret.MediaSearchQuery do
+  @enforce_keys [:source]
+  defstruct [:source, :user, :filter, :q, page: 1]
+end
+
+defmodule Ret.MediaSearchResult do
+  @enforce_keys [:meta, :entries]
+  defstruct [:meta, :entries]
+end
+
+defmodule Ret.MediaSearchResultMeta do
+  @enforce_keys [:source, :page, :page_size, :total_pages, :total_entries]
+  defstruct [:source, :page, :page_size, :total_pages, :total_entries]
+end
+
+defmodule Ret.MediaSearch do
+  import Ret.HttpUtils
+  import Ecto.Query
+
+  alias Ret.{Repo, OwnedFile, SceneListing}
+
+  @page_size 24
+
+  def search(%Ret.MediaSearchQuery{source: "scene_listings", page: page, filter: "featured", q: query}) do
+    scene_listing_search(page, query, "featured", asc: :order)
+  end
+
+  def search(%Ret.MediaSearchQuery{source: "scene_listings", page: page, filter: filter, q: query}) do
+    scene_listing_search(page, query, filter)
+  end
+
+  def search(%Ret.MediaSearchQuery{source: "sketchfab", user: user}) do
+    with api_key when is_binary(api_key) <- resolver_config(:sketchfab_api_key) do
+      res =
+        "https://api.sketchfab.com/v3/search?type=models&downloadable=true&user=#{user}"
+        |> retry_get_until_success([{"Authorization", "Token #{api_key}"}])
```

Comment:
We could limit poly size with a filter here.

Reply:
yep, this is a WIP from a while ago, I'm planning on working on this API next and agree we should be doing some smart stuff here